pxr/usd/usdGeom/testenv/testUsdGeomValidators.cpp

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright 2024 Pixar
+//
+// Licensed under the terms set forth in the LICENSE.txt file available at
+// https://openusd.org/license.
+//
+
+#include "pxr/usd/usd/validator.h"
+#include "pxr/usd/usd/validationError.h"
+#include "pxr/usd/usdGeom/validatorTokens.h"
+#include "pxr/usd/usd/validationRegistry.h"
+#include "pxr/usd/usdGeom/metrics.h"
+#include "pxr/usd/usdGeom/tokens.h"
+
+#include <iostream>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+static
+void TestUsdStageMetadata()
+{
+    // Get stageMetadataChecker
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+    const UsdValidator *validator = registry.GetOrLoadValidatorByName(
+            UsdGeomValidatorNameTokens->stageMetadataChecker);
+    TF_AXIOM(validator);
+
+    // Create an empty stage
+    UsdStageRefPtr usdStage = UsdStage::CreateInMemory();
+
+    UsdValidationErrorVector errors = validator->Validate(usdStage);
+
+    // Verify both metersPerUnit and upAxis errors are present
+    TF_AXIOM(errors.size() == 2);
+    TF_AXIOM(errors[0].GetType() == UsdValidationErrorType::Error);
```

Comment:
Other than GetMessage, you can probably loop to size of errors. And then for GetMessage you can explicitly check the 0th and 1st index?

Reply:
added these messages to their own list as well so everything can be done in the loop. let me know what you think